pkg/storage/fs/ocis/ocis.go

```diff
@@ -207,8 +195,83 @@ func (fs *ocisfs) CreateDir(ctx context.Context, fn string) (err error) {
 	return fs.tp.CreateDir(ctx, node)
 }
 
-func (fs *ocisfs) CreateReference(ctx context.Context, path string, targetURI *url.URL) error {
-	return fs.tp.CreateReference(ctx, path, targetURI)
+// The share folder, aka "/Shares", is a virtual thing.
+func (fs *ocisfs) isShareFolder(ctx context.Context, p string) bool {
+	// TODO what about a folder '/Sharesabc' ... would also match
+	return strings.HasPrefix(p, fs.pw.ShareFolder)
+}
+
+func (fs *ocisfs) isShareFolderRoot(ctx context.Context, p string) bool {
+	return path.Clean(p) == fs.pw.ShareFolder
+}
+
+func (fs *ocisfs) isShareFolderChild(ctx context.Context, p string) bool {
+	p = path.Clean(p)
+	vals := strings.Split(p, fs.pw.ShareFolder+"/")
+	return len(vals) > 1 && vals[1] != ""
+}
+func (fs *ocisfs) getInternalHome(ctx context.Context) (string, error) {
+	if !fs.pw.EnableHome {
+		return "", errtypes.NotSupported("ocisfs: get home not enabled")
+	}
+
+	u, ok := user.ContextGetUser(ctx)
+	if !ok {
+		return "", errtypes.InternalError("ocisfs: wrap: no user in ctx and home is enabled")
+	}
+
+	relativeHome := templates.WithUser(u, fs.pw.UserLayout)
+	return relativeHome, nil
+}
+
+// TODO there really is no difference between the /Shares folder and normal nodes
+// we can store it is a node
+// but when home is enabled the "/Shares" folder should not be listed ...
+// so when listing home we need to check that flag and add the shared node
+
+// /Shares/mounted1
+// /Shares/mounted1/foo/bar
```

Comment:
invalid old example, we can only create references from a path directly under "Shares"